overlord/overlord.go

```diff
@@ -93,6 +101,36 @@ func loadState(backend state.Backend) (*state.State, error) {
 	return state.ReadState(backend, r)
 }
 
+// Run runs a loop to ensure the current state regularly through StateEngine Ensure().
+func (o *Overlord) Run() {
+	intv := ensureInterval
+	o.loopTomb.Go(func() error {
+		tim := time.NewTimer(intv)
+		for {
+			select {
+			case <-o.loopTomb.Dying():
+				return nil
+			case <-tim.C:
+			}
+			err := o.stateEng.Ensure()
+			if err != nil {
+				logger.Panicf("state engine ensure failed not recoverably: %v", err)
+			}
+			tim.Reset(intv)
+		}
+	})
+}
+
+// Stop stops the ensure loop and the managers under the StateEngine.
+func (o *Overlord) Stop() error {
+	o.loopTomb.Kill(nil)
+	err := o.stateEng.Stop()
+	if err != nil {
+		return err
```

Comment:
This means we're potentially returning before the tomb said it was done, which means concurrency in the background doing whatever it wants. We should likely `tomb.Wait` first, then `return engine.Stop`. This will also mean we do not ask for `Stop` in the middle of an `Ensure` call, which feels healthy.